yt_dlp/extractor/beacon.py

```diff
@@ -0,0 +1,102 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    parse_iso8601,
+    traverse_obj,
+)
+
+
+class BeaconTvIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?beacon\.tv/content/(?P<id>.+)'
+
+    _TESTS = [{
+        'url': 'https://beacon.tv/content/welcome-to-beacon',
+        'md5': 'b3f5932d437f288e662f10f3bfc5bd04',
+        'info_dict': {
+            'id': 'welcome-to-beacon',
+            'ext': 'mp4',
+            'upload_date': '20240509',
+            'description': 'md5:ea2bd32e71acf3f9fca6937412cc3563',
+            'thumbnail': 'https://imagedelivery.net/_18WuG-hYMyVquF6KgU_xQ/critrole-beacon-staging-backend/beacon_trailer_thumbnail_beacon_website-5-600x600.png/format=auto',
+            'title': 'Your home for Critical Role!',
+            'timestamp': 1715227200,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        json_data_string = self._html_search_regex(r'<script id="__NEXT_DATA__" type="application/json">(.+)</script>', webpage, 'JSON Body')
+        json_data = self._parse_json(json_data_string, video_id)
+
+        state = traverse_obj(json_data, ('props', 'pageProps', '__APOLLO_STATE__'))
+
+        content_data = None
+        for key, value in state.items():
+            # We can be given many different content objects, we want the one where the slug matches the video ID.
+            if key.startswith('Content') and traverse_obj(value, ('slug')) == video_id:
+                content_data = value
+                break
+
+        # If the user is not authenticated, and this video is not public, the content will be hidden. In this case show an error to the user.
+        if content_data is None:
+            raise ExtractorError('Failed to find content data. Either the given content is not a video, or it requires authentication', expected=True)
+        if content_data['contentVideo'] is None:
+            raise ExtractorError('Failed to find content video. Either the given content is not a video, or it requires authentication', expected=True)
+
```

Comment:
How about we just use the jwplayer data?
```suggestion

        return self._parse_jwplayer_data(traverse_obj(
            content_data, ('contentVideo', 'video', 'videoData', {json.loads})))
```

If you want to include some additional data from the page api you should merge the two info dicts rather than re-implementing the jwplayer code.